### ISSUE TITLE
Razor Precompilation Tooling on migration to MSBuild/csproj failed

### ISSUE THREAD

**Issue by GuardRex** [aspnet/Razor#864]
I just migrated my `netcoreapp1.1` test app (using `dotnet migrate` under `1.0.0-preview4-4079`), which included working Razor precompilation tooling, from `project.json` to MSBuild/`.csproj`. The precompliation tooling became ...

```
<PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design">
  <Version>1.0.0-preview1-final</Version>
  <PrivateAssets>All</PrivateAssets>
</PackageReference>
...
<ItemGroup>
  <DotNetCliToolReference Include="Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools">
    <Version>1.0.0-preview1-final</Version>
  </DotNetCliToolReference>
</ItemGroup>
...
<Target Name="PostpublishScript" AfterTargets="Publish" Condition=" '$(IsCrossTargetingBuild)' != 'true' ">
  <Exec Command="dotnet razor-precompile --configuration $(Configuration) --framework $(TargetFramework) --output-path $(TargetDir) $(MSBuildThisFileDirectory)" />
</Target>
```
Restores fine, but it :boom: on publish ...
```
PostpublishScript:
         dotnet razor-precompile --configuration Release --framework netcoreapp1.1 --output-path x\testselfcontained\bin\Release\netcoreapp1.1\ x\testselfcontained\
         Project 'x\testselfcontained\' does not support framework: netcoreapp1.1
            at Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal.PrecompileDispatchCommand.GetRuntimeContext()
            at Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal.PrecompileDispatchCommand.Execute()
            at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
            at Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design.Internal.PrecompilationApplication.Execute(String[] args)
     1>x\testselfcontained\testselfcontained.csproj(75,5): error MSB3073: The command "dotnet razor-precompile --configuration Release --framework netcoreapp1.1 --output-path x\testselfcontained\bin\Release\netcoreapp1.1\ x\testselfcontained\" exited with code 1.
     1>Done Building Project "x\testselfcontained\testselfcontained.csproj" (Publish target(s)) -- FAILED.

Build FAILED.

       "x\testselfcontained\testselfcontained.csproj" (Publish target) (1) ->
       (PostpublishScript target) -> 
         x\testselfcontained\testselfcontained.csproj(75,5): error MSB3073: The command "dotnet razor-precompile --configuration Release --framework netcoreapp1.1 --output-path x\testselfcontained\bin\Release\netcoreapp1.1\ x\testselfcontained\" exited with code 1.

    0 Warning(s)
    1 Error(s)
```
I've placed the entire `csproj` in a Gist here: https://gist.github.com/GuardRex/58bb9e17a9b372c7bfcd3f79043365ab

```
.NET Command Line Tools (1.0.0-preview4-004079)

Product Information:
 Version:            1.0.0-preview4-004079
 Commit SHA-1 hash:  43dfa6b8ba

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by pranavkm** [comment]
@GuardRex we renamed the package in this release. The packages are now named `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design` and `Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Tools`. We also fixed a bug in it recently where the tool wouldn't work correctly for applications targeting `netcoreapp1.1`: https://github.com/aspnet/MvcPrecompilation/issues/30. Could you try updating and see if that helps?

**Comment by GuardRex** [comment]
Thanks. I'll update and reopen if I'm still having issues.

**Comment by GuardRex** [comment]
@pranavkm The update to `1.1.0-preview4-final` of those two packages seems to have solved that one, but now I have a new one that doesn't seem to match aspnet/MvcPrecompilation#30.

Here's what I have ...
```
PostpublishScript:
dotnet razor-precompile --configuration Release --framework netcoreapp1.1 --output-path <PATH>\testselfcontained\bin\Release\netcoreapp1.1\ <PATH>\testselfcontained\
Could not find file '<PATH>\testselfcontained\project.json'.
     at System.IO.Win32FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
     at System.IO.Win32FileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
     at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
     at Microsoft.DotNet.ProjectModel.ProjectReader.GetProject(String projectPath, ProjectReaderSettings settings)
     at Microsoft.DotNet.ProjectModel.ProjectContext.<CreateContextForEachFramework>d__45.MoveNext()
     at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source, Int32& length)
     at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
     at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.PrecompileDispatchCommand.GetRuntimeContext()
     at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Internal.PrecompileDispatchCommand.Execute()
     at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
     at Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design.Internal.PrecompilationApplication.Execute(String[] args)
     1><PATH>\testselfcontained\testselfcontained.csproj(78,5): error MSB3073: The command "dotnet razor-precompile --configuration Release --framework netcoreapp1.1 --output-path <PATH>\testselfcontained\bin\Release\netcoreapp1.1\ <PATH>\testselfcontained\" exited with code 1.
```

**Comment by pranavkm** [comment]
We did not intend for the 1.1.0 versions of tooling packages to work with MsBuild. There's plans to release a set of updates to our tools that we released as part of 1.0.0 to support MSBuild based projects. However since Razor compilation was introduced as part of the 1.1.0, we did not do any work there (primarily because it requires AspNetCore Mvc 1.1.0).
